enterprise/src/main/java/io/crate/enterprise/loader/InstantiableClass.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.enterprise.loader;
+
+/**
+ * A class which can be loaded and instantiated during runtime.
+ * @param <T> The super class type of the instantiated class
+ */
+public interface InstantiableClass<T> extends LoadableClass<T> {
```

Comment:
I thought about using method parameters. I came to the conclusion that encapsulating the loading information in classes makes it more readable. If we got rid of enterprise setting related parameters in the method, it might be more feasible.